syntax = "proto3";
import "google/protobuf/struct.proto";

service MicroserviceController {
  rpc Ping(PingMessage) returns (PingResponse);
  rpc StartService(StartAndStopCommand) returns (Result);
  rpc StopService(StartAndStopCommand) returns (Result);
}

message PingMessage {
  string client_id = 1;
}

message PingResponse {
  google.protobuf.Struct data = 1;
}

message StartAndStopCommand {
  string client_id = 1;
}

message Result{
  bool success = 1;
}